spec/lib/synapse/haproxy_spec.rb

```diff
@@ -21,7 +21,18 @@ class MockWatcher; end;
   let(:mockwatcher_with_server_options) do
     mockWatcher = double(Synapse::ServiceWatcher)
     allow(mockWatcher).to receive(:name).and_return('example_service2')
-    backends = [{ 'host' => 'somehost', 'port' => 5555, 'haproxy_server_options' => 'backup'}]
+    backends = [{ 'host' => 'somehost', 'port' => 5555, 'haproxy_server_options' => 'id 12 backup'}]
```

Comment:
Could you add a test with several backends and one setting 'haproxy_server_options' => 'id <max_server_id>'